include/aws/s3/private/s3_buffer_pool.h

```diff
@@ -111,6 +121,17 @@ AWS_S3_API struct aws_byte_buf aws_s3_buffer_pool_acquire_buffer(
     struct aws_s3_buffer_pool *buffer_pool,
     struct aws_s3_buffer_pool_ticket *ticket);
 
+/*
+ * Force immediate acquisition of a buffer from the pool.
+ * This should only be used if waiting to reserve a ticket would risk deadlock.
+ * This cannot fail, not even if the pool has a reservation hold,
+ * not even if the memory limit has been exceeded.
+ */
+AWS_S3_API struct aws_byte_buf aws_s3_buffer_pool_acquire_forced_buffer(
```

Comment:
not having it be able to fail seems dangerous. we should probably plan for supporting some sort of bound on forced allocations (if not now than in future). we should have an option for client to enforce how much mem can be used by forced allocations